src/main/java/uk/gov/hmcts/reform/idam/service/UserRoleService.java

```diff
@@ -59,4 +57,84 @@ public List<String> filterUsersWithRoles(List<String> staleUsers) {
+        var response = roleAssignmentClient.getRoleAssignmentsByUserId(securityUtil.getAuthHeaders(), archivedUserId);
+
+        RoleAssignmentsMergeRequest mergeRequest = createMergeRequest(response, archivedUserId, activeUserId);
+        Map<String, String> headers = securityUtil.getAuthHeaders();
+        try (var mergeResponse = roleAssignmentClient.createRoleAssignment(headers, mergeRequest)) {
+            if (mergeResponse.status() >= 300) {
+                duplicateUserSummary.increaseFailedMerge();
+                log.error("Merge failed for user active user {} (deleted user id {})", activeUserId, archivedUserId);
+            }
+        }
+    }
+
+    private RoleAssignmentsMergeRequest createMergeRequest(
+        RoleAssignmentResponse response,
+        String archivedUserId,
+        String activeUserId
+    ) {
+        List<RoleAssignment> roleAssignments = response.getRoleAssignments();
+        String caseId = checkRoleAssignments(roleAssignments, archivedUserId);
+        String reference = String.format("%s-%s", caseId, activeUserId);
+
+        RoleRequest roleRequest = RoleRequest.builder()
+            .requestType("CREATE")
+            .process("CCD")
+            .reference(reference)
+            .assignerId(activeUserId) // ???
+            .replaceExisting(false)
+            .build();
+
+        List<RequestedRole> requestedRoles = new LinkedList<>();
+        for (var roleAssignment: roleAssignments) {
+            requestedRoles.add(buildRequestedRole(roleAssignment, activeUserId));
+        }
+
+        return RoleAssignmentsMergeRequest.builder()
+            .roleRequest(roleRequest)
+            .requestedRoles(requestedRoles)
+            .build();
+    }
+
+    private RequestedRole buildRequestedRole(RoleAssignment roleAssignment, String activeUserId) {
+        return RequestedRole.builder()
+            .actorIdType(roleAssignment.getActorIdType())
+            .actorId(activeUserId)
+            .roleType(roleAssignment.getRoleType())
+            .roleName(roleAssignment.getRoleName())
+            .classification(roleAssignment.getClassification())
+            .grantType(roleAssignment.getGrantType())
+            .roleCategory(roleAssignment.getRoleCategory())
+            .readOnly(roleAssignment.isReadOnly())
+            .beginTime(roleAssignment.getBeginTime())
+            .process("CCD")
+            .attributes(roleAssignment.getAttributes())
+            .build();
+    }
+
+    private String checkRoleAssignments(List<RoleAssignment> roleAssignments, String archivedUserId) {
+        var caseIds = roleAssignments.stream()
+            .map(RoleAssignment::getAttributes)
+            .map(RoleAssignmentAttributes::getCaseId)
+            .collect(Collectors.toSet());
+
+        if (caseIds.size() > 1) {
+            throw new UnsupportedOperationException("Different case ids found on a single user " + archivedUserId);
+        }
+
+        if (caseIds.isEmpty()) {
```

Comment:
We could add this to the summary

Reply:
At the moment it throws an exception, meaning we crash the application. There shouldn't be any such cases and if we find it, it is possible because there was a mistake somewhere in our code. That is the reason I've decided to do this way. I can change it to ignore this scenario and simply put into stats that that has happened.